geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,411 @@
+# #  Copyright (c) 2024 Mira Geoscience Ltd.
+# #
+# #  This file is part of geoh5py.
+# #
+# #  geoh5py is free software: you can redistribute it and/or modify
+# #  it under the terms of the GNU Lesser General Public License as published by
+# #  the Free Software Foundation, either version 3 of the License, or
+# #  (at your option) any later version.
+# #
+# #  geoh5py is distributed in the hope that it will be useful,
+# #  but WITHOUT ANY WARRANTY; without even the implied warranty of
+# #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# #  GNU Lesser General Public License for more details.
+# #
+# #  You should have received a copy of the GNU Lesser General Public License
+# #  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+#
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+from uuid import UUID
+
+from pydantic import BaseModel, model_validator
+
+from ..shared.utils import str2uuid
+from .base import Group
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from ..objects import Curve, Slicer
+
+ATTRIBUTE_MAP = {
+    "Normal X": "normal_x",
+    "Normal Y": "normal_y",
+    "Normal Z": "normal_z",
+    "Position X": "position_x",
+    "Position Y": "position_y",
+    "Position Z": "position_z",
+}
+
+
+class InterpretationSectionParams(BaseModel):
+    """
+    The class to store the interpretation section.
+
+    :param normal_x: The normal x component.
+    :param normal_y: The normal y component.
+    :param normal_z: The normal z component.
+    :param position_x: The position x component.
+    :param position_y: The position y component.
+    :param position_z: The position z component.
+    """
+
+    normal_x: float
+    normal_y: float
+    normal_z: float
+    position_x: float
+    position_y: float
+    position_z: float
+
+    def in_list(self, params_list: list[InterpretationSectionParams]) -> int | None:
+        """
+        Check if the parameters are in the list.
+
+        :param params_list: The list of parameters.
+
+        :return: The index of the parameters in the list else None.
+        """
+        for idx, param in enumerate(params_list):
+            if param.map_back == self.map_back:
+                return idx
+        return None
```

Comment:
This is pretty much the same as doing `self in params_list`. I don't think it makes sense to have this logic at the Section level, but rather query the `self._interpretation_sections`

Reply:
I guess I don't see the use case of removing a section by using parameter values - it is both very unrealiable (float comparison is always tricky) and not very practical. Makes sense to let users remove one from the list though.